synapse/models/infotech.py

```diff
@@ -303,6 +462,46 @@ def getBaseModels():
                     ('reg:int', {'ptype': 'it:dev:int', 'doc': 'The integer value that was deleted (parsed from :reg).', 'ro': 1}),
                     ('reg:bytes', {'ptype': 'file:bytes', 'doc': 'The binary data that was deleted (parsed from :reg).', 'ro': 1}),
                 )),
+                ('it:prod:soft', {}, (
+                    ('name', {'ptype': 'str:lwr', 'ro': 1, 'req': 1,
+                              'doc': 'Name of the software'}),
+                    ('desc', {'ptype': 'str:txt', 'doc': 'A description of the software'}),
+                    ('desc:short', {'ptype': 'str:lwr', 'doc': 'A short description of the software'}),
+                    ('author:org', {'ptype': 'ou:org', 'doc': 'Organization responsible for the software', }),
+                    ('author:acct', {'ptype': 'inet:web:acct', 'doc': 'Web user responsible for the software', }),
+                    ('author:person', {'ptype': 'ps:person', 'doc': 'Person responsible for the software', }),
+                    ('url', {'ptype': 'inet:url', 'doc': 'URL relevant for the software', }),
+                )),
+
+                ('it:prod:softver', {}, (
```

Comment:
Thinking out loud...one way to track "code families" (simplified "it:prod:soft / it:prod:softver") is via tags (e.g., "code.psexec"). I expect in many cases (for simplicity) we'll still want to do this, but will also want the ability to have a relationship node that says (for simple programs, at least) "this file:bytes is psexec version 2.5". (Just a comment, not a fix...)